float64 width
---
bool success